name: Auto Merge Owner PR1

on:
  pull_request:
    types: [opened, reopened]

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    
    - uses: actions/checkout@v1
    
    - uses: actions/setup-python@v5 
      with:
        python-version: 'pypy3.10'
    
    - name: flake8 and pylint
      run: |
        pip install -r requirements.txt
        flake8 --ignore=E501,E231 *.py tests/*.py
        pylint --errors-only --disable=C0301 --disable=C0326 *.py tests/*.py
    
    - name: unit test
      run: python -m unittest --verbose --failfast
  
  merge:
    if: github.actor == 'rajeshnaik2020' || github.actor == 'rnmsazure' ||  github.actor == 'dependabot[bot]' || github.actor == 'github-actions[bot]'
    needs: [lint]
    runs-on: ubuntu-latest
    permissions:
      pull-requests: write
    steps:
    
    - uses: actions/checkout@v1
    
    - uses: hmarr/auto-approve-action@v4.0.0
      with:
        github-token: ${{ secrets.GITHUB_TOKEN }}
    
    - uses: managedkaos/merge-pull-request@v2.2.0
      env:
        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
